src/lib/djerba/helpers/pwgs_provenance_helper/helper.py

```diff
@@ -0,0 +1,162 @@
+        """
+        config = self.apply_defaults(config)
+        wrapper = self.get_config_wrapper(config)
+        provenance_path = wrapper.get_my_string(self.PROVENANCE_INPUT_KEY)
+        work_dir = self.workspace.get_work_dir()
+        # Set INI parameters from cardea, if available
+        if os.path.exists(os.path.join(work_dir,core_constants.DEFAULT_SAMPLE_INFO)):
+            input_data = self.workspace.read_json(core_constants.DEFAULT_SAMPLE_INFO)
+            if wrapper.my_param_is_null(pc.PROJECT):
+                wrapper.set_my_param(pc.PROJECT, input_data[pc.PROJECT])
+            if wrapper.my_param_is_null(pc.DONOR):
+                wrapper.set_my_param(pc.DONOR, input_data[pc.DONOR])
+            if wrapper.my_param_is_null(pc.PROVENANCE_ID):
+                wrapper.set_my_param(pc.PROVENANCE_ID, input_data[pc.PROVENANCE_ID])
+        else:
+            msg = "Sample Info JSON does not exist. Parameters must be set manually."
+            self.logger.warning(msg)
+        project = wrapper.get_my_string(pc.PROJECT) 
+        donor = wrapper.get_my_string(pc.DONOR)
+        # Write Provenance subset, if it doesn't exist
+        if self.workspace.has_file(self.PROVENANCE_OUTPUT):
+            cache_path = self.workspace.abs_path(self.PROVENANCE_OUTPUT)
+            msg = "Provenance subset cache {0} exists, will not overwrite".format(cache_path)
+            self.logger.info(msg)
+        else:
+            self.logger.info("Writing provenance subset cache to workspace")
+            self.write_provenance_subset(project, donor, provenance_path)
+        # Make path_info file, if it doesn't exist
+        if self.workspace.has_file(core_constants.DEFAULT_PATH_INFO):
+            msg = "extract: path info files already in workspace, will not overwrite"
+            self.logger.info(msg)
+        else:
+            mrdetect_file_suffixes = [pc.RESULTS_SUFFIX, pc.VAF_SUFFIX, pc.HBC_SUFFIX, pc.SNV_COUNT_SUFFIX]
+            path_info = self.subset_provenance("mrdetect", wrapper.get_my_string(pc.PROVENANCE_ID), mrdetect_file_suffixes)
+            path_info.update(self.subset_provenance("dnaSeqQC",  wrapper.get_my_string(pc.PROVENANCE_ID), [pc.BAMQC_SUFFIX]))
+            self.write_path_info(path_info)
+        return wrapper.get_config()
+
+    def extract(self, config):
+        self.validate_full_config(config)
+    
+    def parse_file_path(self, file_pattern, provenance):
+        # get most recent file of given file path pattern,
+        iterrows = self._filter_file_path(file_pattern, rows=provenance)
+        try:
+            row = self._get_most_recent_row(iterrows)
+            path = row[index.FILE_PATH]
+        except MissingProvenanceError as err:
+            msg = "No provenance records meet filter criteria: path-regex = {0}.".format(file_pattern)
+            self.logger.debug(msg)
+            path = None
+        return path
+
+    def subset_provenance(self, workflow, provenance_id, suffixes):
```

Comment:
Maybe rename this to something like `read_provenance_subset`? That would make it clearer that it's reading the subset file, not the complete file provenance, and have a nice symmetry with `write_provenance_subset`.